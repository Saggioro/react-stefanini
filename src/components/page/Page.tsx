import React, { FunctionComponent } from 'react';
import Header from '../header/Header';

import {Container} from './Page.styled';
interface IPage {
    children: React.ReactNode;
}

const Page:FunctionComponent<IPage> = ({children, ...props}:IPage) => {
    return (
        <Container>
            <Header />
            {children}
        </Container>
    );
};

export default Page;