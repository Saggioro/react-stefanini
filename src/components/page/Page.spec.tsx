/* eslint-disable testing-library/prefer-screen-queries */
/* eslint-disable testing-library/no-debugging-utils */
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import Page from "./Page";
jest.mock("react-redux", () => {
  return {
    useDispatch() {
      return {
        dispatch: (user: any) => {
          user = undefined;
        },
      };
    },
  };
});

jest.mock("../../hooks/LocalReduxThunk", () => {
  return {
    useAppSelector() {
      return {
        name: "admin teste",
        password: "123",
      };
    },
  };
});

test("Input should render correctly", () => {
  const component = renderer.create(<Page>content</Page>);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Page label should render content", () => {
  render(<Page>content</Page>);

  expect(screen.getByText("content")).toBeInTheDocument();
});
