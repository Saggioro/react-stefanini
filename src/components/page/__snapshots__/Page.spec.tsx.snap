// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<div
  className="sc-hKMtZM ifVoBq"
>
  <nav
    className="sc-bczRLJ fKaIsd"
  >
    <h2
      className="sc-dkzDqf hDIAhY"
    >
      Hello, 
      admin
    </h2>
    <button
      className="sc-gsnTZi bicnmt"
      onClick={[Function]}
    >
      Logout
    </button>
  </nav>
  content
</div>
`;
